src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/EqualsAvoidNullCheck.java

```diff
@@ -444,9 +444,7 @@ private boolean isStringFieldOrVariableFromClass(DetailAST objCalledOn,
         while (frame != null) {
             if (className.equals(frame.getFrameName())) {
                 final DetailAST field = frame.findField(name);
-                if (field != null) {
-                    result = STRING.equals(getFieldType(field));
-                }
+                result = STRING.equals(getFieldType(field));
```

Comment:
This line removed to reach 100% coverage.

Reply:
Pitest or coverage ?
Looks suspicious, please explain why null is not possible anymore.

Reply:
This is only for classes defined in the current file. If a file can be compiled, all fields are guaranteed to be found. Before this PR there was a bug causing a call to
```
frame.findField("((fieldName))")
```
which returns `null`.  Changes in the method `skipVariableAssign` fixes this.

Reply:
Now it explains all other code, I moved comment to description